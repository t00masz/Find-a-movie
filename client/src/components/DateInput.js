import React from 'react';
import changeUserData from '../actions/user-actions';
import { connect } from 'react-redux';
import moment from 'moment';

const DateInput = (props) => (
  <div className="option">
    <input name={props.name} 
      className="option__input" 
      type="date" 
      value={props.value} 
      required
      min={moment().format("YYYY-MM-DD")}
      onChange = {(e) => {
        props.dispatch(changeUserData(props.name, e.target.value))
      }
    }/>    
  </div>
);

export default connect()(DateInput);