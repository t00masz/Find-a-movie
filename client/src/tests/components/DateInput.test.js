import React from 'react';
import { shallow } from 'enzyme'
import DateInput from '../../components/Label';
import user from '../fixtures/user';

test('should render date input', () => {
    const wrapper = shallow(<DateInput 
        name="date"
        value={user.date}/>);
    expect(wrapper).toMatchSnapshot();
});
