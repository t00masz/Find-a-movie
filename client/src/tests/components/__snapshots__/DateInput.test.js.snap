// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render date input 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Label
    name="date"
    value="1970-01-01"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <label
        className="option__label"
        name="date"
>
        1970-01-01
</label>,
      "className": "option",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "1970-01-01",
        "className": "option__label",
        "name": "date",
      },
      "ref": null,
      "rendered": "1970-01-01",
      "type": "label",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <label
          className="option__label"
          name="date"
>
          1970-01-01
</label>,
        "className": "option",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "1970-01-01",
          "className": "option__label",
          "name": "date",
        },
        "ref": null,
        "rendered": "1970-01-01",
        "type": "label",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
